core/controllers/resources_test.py

```diff
@@ -223,8 +223,18 @@ def test_bad_extensions_are_detected(self):
 
         self.logout()
 
+    def test_request_invalid_asset_type(self):
+        """Test that requests for invalid asset type is rejected with a 404."""
+        self.login(self.EDITOR_EMAIL)
+
+        response = self.testapp.get(
+            '/assetsdevhandler/0/assets/unknown/myfile',
```

Comment:
Maybe for clarity of reading, put unknown_type here (instead of unknown).

Also I think you should immediately follow this with a check for /0/assets/image/... so that it's clear that the 404 is not due to something else.

Reply:
Coming to think of it, I'm not sure the test make sense because both calling an unsupported type and calling a supported type but non existing file would cause a 404 error.

Feel free to let me know if you have any suggestions.

Reply:
I was referring to test_get_invalid_image above, which just checks a 404 from a non existing image.

Reply:
For this test, you could add the image before making the two calls?

Reply:
Oh, actually -- yeah, I see what you mean. The unknown type doesn't really have any namespace of files corresponding to it anyway.

I think maybe it's fine as is, then.